### PR TITLE
Enable Github Actions for building

### PR DESCRIPTION
**Pull request by mmcky**
@orectique can you please review this build

https://64fc037ed0af9f2eff6bc494--poetic-sprite-e126a3.netlify.app/jl/introductions

it has some missing packages like `Distributions` for `jl`?

### CODE REVIEW

**Comment by mmcky** [comment]
@orectique please feel free to push directly to this branch.

**Comment by orectique** [comment]
> @orectique can you please review this build
> 
> https://64fc037ed0af9f2eff6bc494--poetic-sprite-e126a3.netlify.app/jl/introductions
> 
> it has some missing packages like `Distributions` for `jl`?

I am facing that on my local Linux version as well. Will figure it out.

I'll add LaTeX libraries to the environment.yml files as well.

**Comment by mmcky** [comment]
> I'll add LaTeX libraries to the environment.yml file as well.

Best to uncomment the LaTeX installs in `.github/workflows/ci.yml` if you need LaTeX support on the build system

**Comment by orectique** [comment]
> @orectique please feel free to push directly to this branch.

@mmcky I do not have permissions to push to this repo.

**Comment by mmcky** [comment]
@orectique thanks. 

Are you still working on the following issue:

```
PyError ($(Expr(:escape, :(ccall(#= /home/runner/.julia/packages/PyCall/ilqDX/src/pyfncall.jl:43 =# @pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, pyargsptr, kw))))) <class 'FileNotFoundError'>
```

**Comment by mmcky** [comment]
@orectique not sure what is going on with `PyCall` -- I have installed it, linked it to python but no luck in getting the PyCall fixed so far.

**Comment by mmcky** [comment]
@orectique I would recommend the following workflow:

**Step 1:** Build a Jupyter book Project from the `source` files
**Step 2:** Run `jupyter-book` over the project to build `html` etc. 

That would reduce the complexity in diagnosing this issue. 

If you have time this afternoon to make this refactor that would be great. @jstac needs this site up and running by his morning (US time) so I have some time tomorrow morning to integrate into the textbook site.

**Comment by orectique** [comment]
PyCall has been giving me trouble as well. 

I'll split the script into two right away.

**Comment by mmcky** [comment]
Nice work @orectique. 

When you get time (not urgent) can you document what the main issues were around `julia` and `pycall` and I might add that as configuration into the `quantecon` manual to retain this knowledge.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      #       cm-super                  \
+      #       msttcorefonts
+      - name: Set up Julia
+        uses: julia-actions/setup-julia@v1
+        with:
+          version: 1.9.3
+      - name: Install IJulia and Setup Project
+        shell: bash
+        run: |
+          julia -e 'using Pkg; ENV["PYTHON"]="/usr/share/miniconda3"; Pkg.instantiate()'
```
Comment (mmcky) [inline .github/workflows/publish.yml:45]: @orectique should this be update to include the `conda` installs etc. that you have added to `ci.yml`?

**Comment by mmcky** [comment]
@orectique can you confirm with the two `py` files do we only need to run

```
python create_code_book.py
```

or do we need to add `python create_jb.py`? 

Thanks for confirmation. 

- [x] I am about to push updates to `publish.yml` to mirror the Julia updates you added to `ci.yml`

File: .github/workflows/publish.yml (hunk 1 above)
Comment (orectique) [reply .github/workflows/publish.yml]: Hi @mmcky

I see you have already updated `publish.yml`. Thank you

**Comment by orectique** [comment]
@mmcky I've split `create_code_book.py` into two parts - `create_jb.py` (which compiles the scripts into the JB format - to be run when the scripts are updated) and `build_jb.py` (which basically runs jb build). 

`create_jb.py` **does not** build the book. Both scripts will have to be called in succession.

The current workflow uses `create_code_book.py`. We can swap it out for the two scripts if you'd prefer that.

**Comment by mmcky** [comment]
thanks @orectique no I just wanted to make sure I was building things correctly on `GitHub actions`. I think in the future we will revert to calling jupyter book directly on the code-book files but this works well for now.

**Comment by mmcky** [comment]
@orectique once this passes are you happy for me to merge?

**Comment by orectique** [comment]
@mmcky yep. I think we can go ahead and merge.

**Pull request merged**